events/importer/lippupiste.py

```diff
@@ -0,0 +1,371 @@
+    text = clean_text(text)
+    return text
+
+
+def clean_short_description(text):
+    text = replace_html_breaks_with_whitespace(text)
+    text = strip_tags(text)
+    text = clean_text(text)
+    if '.' in text:
+        text = text.split('.')
+        text = text[0] + '.'
+    text = text[:160]
+    return text
+
+
+def get_namespaced_event_serie_id(event_serie_id):
+    return 'serie-%s' % event_serie_id
+
+
+@register_importer
+class LippupisteImporter(Importer):
+    name = 'lippupiste'
+    supported_languages = ['fi']
+
+    def _cache_yso_keyword_objects(self):
+        try:
+            yso_data_source = DataSource.objects.get(id='yso')
+        except DataSource.DoesNotExist:
+            self.keyword_by_id = {}
+            return
+        keyword_id_set = set()
+        for yso_keyword_ids in YSO_KEYWORD_MAPS.values():
+            for keyword_id in yso_keyword_ids:
+                keyword_id_set.add(keyword_id)
+        keyword_list = Keyword.objects.filter(data_source=yso_data_source).filter(id__in=keyword_id_set)
+        self.keyword_by_id = {keyword.id: keyword for keyword in keyword_list}
+
+    def _cache_place_data(self):
+        self.place_data_list = Place.objects.filter(data_source=self.tprek_data_source).values(
+            'id', 'name', 'street_address_fi', 'street_address_sv', 'postal_code'
+        )
+        for place_data in self.place_data_list:
+            place_data['name__lower'] = place_data['name'].lower()
+
+            if place_data['street_address_fi']:
+                place_data['street_address_fi__lower'] = place_data['street_address_fi'].lower()
+            else:
+                place_data['street_address_fi__lower'] = None
+
+            if place_data['street_address_sv']:
+                place_data['street_address_sv__lower'] = place_data['street_address_sv'].lower()
+            else:
+                place_data['street_address_sv__lower'] = None
+
+        self.existing_place_id_matches = {}
+
+    def _cache_super_event_ids(self):
+        qs = Event.objects.filter(data_source=self.data_source, super_event_type__isnull=False)
+        self.super_event_ids_by_origin_id = {super_event.origin_id: super_event.id for super_event in qs}
+
+    def setup(self):
+        data_source_args = dict(id=self.name)
+        data_source_defaults = dict(name="Lippupiste")
+        self.data_source, _ = DataSource.objects.get_or_create(defaults=data_source_defaults, **data_source_args)
+        self.tprek_data_source = DataSource.objects.get(id='tprek')
+
+        ytj_data_source, _ = DataSource.objects.get_or_create(defaults={'name': "YTJ"}, id='ytj')
+        org_args = dict(origin_id='1789232-4', data_source=ytj_data_source, internal_type=Organization.AFFILIATED)
+        org_defaults = dict(name="Lippupiste Oy")
+        self.organization, _ = Organization.objects.get_or_create(defaults=org_defaults, **org_args)
+        self._cache_yso_keyword_objects()
```

Comment:
Oh, and for completeness' sake, let's tie the lippupiste data source and organization together by saving the information `self.data_source.owner=self.organization` here, after creating both.